src/main/java/nl/novi/serviceconnect/services/TransactionService.java

```diff
@@ -0,0 +1,118 @@
+package nl.novi.serviceconnect.services;
+
+import nl.novi.serviceconnect.dtos.TransactionInputDto;
+import nl.novi.serviceconnect.dtos.TransactionOutputDto;
+import nl.novi.serviceconnect.exceptions.RecordNotFoundException;
+import nl.novi.serviceconnect.helpper.Helpers;
+import nl.novi.serviceconnect.models.ServiceRequest;
+import nl.novi.serviceconnect.models.Transaction;
+import nl.novi.serviceconnect.repository.ServiceRequestRepository;
+import nl.novi.serviceconnect.repository.TransactionRepository;
+import org.springframework.stereotype.Service;
+import nl.novi.serviceconnect.helpper.Mapper;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TransactionService implements ITransactionService {
+    private final TransactionRepository transactionRepository;
+
+    private final ServiceRequestRepository requestRepository;
+    public TransactionService(TransactionRepository repository, ServiceRequestRepository requestRepository) {
+        this.transactionRepository = repository;
+        this.requestRepository = requestRepository;
+    }
+
+    private String saveFile() {
+        try {
+            File targetDirectory = new File("C:\\Java-HBO\\LesOpdrachten\\EindProject\\ServiceConnect\\pdfs");
+
```

Comment:
Zou je kunnen kijken of je de paden ook flexibeler kunt maken? Op deze manier runt het alleen op jouw systeem, zoals hoe je het nu hebt ingericht.

Reply:
deze is ook opgelost in de volgende PR van transactions en Categories